### PR TITLE
Refactor evaluated values out of entity objects

### PR DESCRIPTION
**Pull request by ofpiyush**
> Pull Request Template
>
> Use this template to quickly create a well written pull request. Delete all quotes before creating the pull request.

## Description
While trying to refactor, I learnt that it might be much bigger in scope than we initially anticipated.
This PR might be a good place to discuss the scope of that refactor and whether we should dive into it

Fixes # (issue)
> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce. 
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/workers/evaluation.worker.ts [hunk 1]
```diff
@@ -61,9 +118,17 @@ ctx.addEventListener("message", e => {
       const response = getEvaluatedDataTree(dataTree);
       // We need to clean it to remove any possible functions inside the tree.
       // If functions exist, it will crash the web worker
+
+      console.log("everything", response, EVALUATED_VALUES.getAll());
       const cleanDataTree = JSON.stringify(response);
```
Comment (hetunandu) [inline app/client/src/workers/evaluation.worker.ts:123]: We do not need to stringify this anymore mostly. We can get away with just the evaluated values to be sanitised.

File: app/client/src/workers/evaluation.worker.ts [hunk 2]
```diff
@@ -270,7 +335,7 @@ const removeFunctionsFromDataTree = (dataTree: DataTree) => {
 const removeFunctions = (value: any) => {
   if (_.isFunction(value)) {
     return "Function call";
-  } else if (_.isObject(value) && _.some(value, _.isFunction)) {
```
Comment (hetunandu) [inline app/client/src/workers/evaluation.worker.ts:273]: Let us keep this only for functions again

File: app/client/src/workers/evaluation.worker.ts [hunk 3]
```diff
@@ -48,8 +48,65 @@ import {
 
 const ctx: Worker = self as any;
 
+type EvaluatedValuesMap = Record<string, any>;
 let ERRORS: EvalError[] = [];
 let WIDGET_TYPE_CONFIG_MAP: WidgetTypeConfigMap = {};
+const EVALUATED_VALUES = new (class {
```
Comment (hetunandu) [inline app/client/src/workers/evaluation.worker.ts:54]: Extend [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) class instead?

File: app/client/src/workers/evaluation.worker.ts [hunk 4]
```diff
@@ -534,6 +600,8 @@ function dependencySortedEvaluateDataTree(
                 widgetEntity,
               );
             }
+            // Todo: (piyush) figure how to decouple this.
```
Comment (ofpiyush) [inline app/client/src/workers/evaluation.worker.ts:603]: @hetunandu this is what I wanted you to see

File: app/client/src/workers/evaluation.worker.ts [hunk 5]
```diff
@@ -614,6 +679,7 @@ const getValidatedTree = (tree: any) => {
             entity,
             tree,
           );
+          // Todo: (piyush) see if we can refactor this out.
```
Comment (ofpiyush) [inline app/client/src/workers/evaluation.worker.ts:682]: @hetunandu and this

**Pull request closed without merging**